lib/metanorma/compile_validate.rb

```diff
@@ -29,31 +29,30 @@ def load_flavor(stdtype)
         Util.log("[metanorma] Info: Loading `#{flavor}` gem "\
                  "for standard type `#{stdtype}`.", :info)
       end
-      require_flavor(flavor, stdtype)
+      require_flavor(flavor)
       unless @registry.supported_backends.include? stdtype
-        Util.log("[metanorma] Error: The `#{flavor}` gem "\
-                 "still doesn't support `#{stdtype}`. Exiting.", :fatal)
+        Util.log("[metanorma] Error: The `#{flavor}` gem does not "\
+                 "support the standard type #{stdtype}. Exiting.", :fatal)
       end
     end
 
-    def require_flavor(flavor, stdtype)
+    def require_flavor(flavor)
       require flavor
       Util.log("[metanorma] Info: gem `#{flavor}` loaded.", :info)
-    rescue Gem::ConflictError
-      Util.log("[metanorma] Error: Couldn't resolve dependencies for "\
-               "`metanorma-#{stdtype}`, Please add it to your Gemfile "\
-               "and run bundle install first", :fatal)
-    rescue LoadError
+    rescue LoadError => e
       msg = <<~MSG
-        [metanorma] Error: loading gem `#{flavor}` failed. Exiting.
+        Error: #{e.message}
+        Metanorma has encountered an exception.
 
-        Troubleshooting:
-        1. If you are using metanorma via bundler/ruby, make sure that your
-           Gemfile contains a line:
-             gem "metanorma-#{stdtype}"
+        If this problem persists, please report this issue at the following link:
 
-        2. If you are using brew/choco/snap packages, please report an issue
-           to https://github.com/metanorma/packed-mn/issues/new"
+        * https://github.com/metanorma/metanorma/issues/new
+
+        We need to also generate the #{Date.today}-error.log file that includes the error message and the error backtrace. Thanks!
```

Comment:
@CAMOBAP this is not what I mean. This line was intended for you... I mean that Metanorma needs to implement functionality to generate the error log 😅

Reply:
@ronaldtse done as proposed but as far as I see the error message is too generic for this particular error (LoadError) because it happens only for unsupported flavors

I propose to:
1. Move this error message to some generic error handler `at_exit` or something like this
2. Use logger from `metanorma-utils` to create some ~/.metanorma/last-run.log and generate error.log which will include not only stack trace but previous log messages too

@ronaldtse @opoudjis any feedback?

Reply:
> I propose to:
> 
> 1. Move this error message to some generic error handler `at_exit` or something like this
> 2. Use logger from `metanorma-utils` to create some ~/.metanorma/last-run.log and generate error.log which will include not only stack trace but previous log messages too

Agree with both!! This is the whole point. Thank you.

(I do have a concern that `metanorma-utils` is now a strange collection of supplementary code, but that's a separate issue)